storm-core/src/jvm/org/apache/storm/trident/TridentTopology.java

```diff
@@ -277,6 +277,35 @@ public Stream join(List<Stream> streams, List<Fields> joinFields, Fields outFiel
               outFields);
     }
 
+    public Stream joinSQL(Stream s1, Fields joinFields1, Stream s2, Fields joinFields2, Fields outFields) {
```

Comment:
@arunmahadevan 
This provides `join` with preserving parameters. I agree that `joinSQL` is bad. I just haven't found any good naming, since `joinWithPreservingParameters` is bad, too. ;) I would like to have good alternative name, maybe `join` with `mode` parameter would be also fine for me.

Btw, implementing this to outside of storm-core is not easy since it relies on existing implementations which is not open to public. Lots of duplications or exposing unneeded things will be required if we want to move this to SQL runtime.